ch_backup/cli.py

```diff
@@ -60,6 +60,28 @@ def signal_handler(signum, _frame):
 signal.signal(signal.SIGINT, signal_handler)
 
 
+class JsonParamType(ParamType):
+    """
+    JsonParamType type for command-line parameter for JSON value.
+    """
+
+    name = "json"
+
+    def convert(self, value, param, ctx):
+        try:
+            if re.fullmatch(
+                r'\s*([\[{"].*|true|false|null|\d+(\.\d+)?)\s*',
+                value,
+                re.MULTILINE | re.DOTALL,
+            ):
+                return json.loads(value)
+            return value.strip()
```

Comment:
How about returning dictionaries always (do parsing in that class). And just merge it (recursively) in the command handler if needed ? It looks like a better encapsulation.

Reply:
But we are returning dictionary right now `return json.loads(value)` except when json is plain value/number/bool.

New parameter `--config-parameter` could be called multiple times at once.
And we can not _easily_ build one dict from several inputs here.

Do you mean create extended ParamType like MultipleJsonParamType which takes several inputs and do all logic from `_build_cli_cfg_from_config_parameters`?
Not sure about such work in this task :)
But for sure it will be easier to add same logic for other parameters in future.

Reply:
I also meant that when `JsonParamType` returns something that is not a dictionary it is a bit confusing. We could process one pair `--config-parameter full_path value`, and always return a dictionary like `{'path1': {'path2': value}}` in custom type. The `click` will process and put to a list several parameters itself.

And then call `config = merge(config_parameters)` in command handler to merge them all. 
But yes for this we need to move part of building logic to that class. 
As for me this is not a blocker, and if you think this is not rational, we can close this ticket.

Reply:
Now I've got it, good idea in future.
It should works with both parameters - `full_path` and `value`, JsonParamType takes only one here.

So right now I will keep current behaviour.